src/main/java/org/jetbrains/research/kotlinrminer/decomposition/OperationBody.java

```diff
@@ -66,14 +66,28 @@ private void processStatement(KtFile ktFile, String filePath, CompositeStatement
             }
         } else if (statement instanceof KtForExpression) {
             KtForExpression forStatement = (KtForExpression) statement;
-            CompositeStatementObject child = new CompositeStatementObject(ktFile, filePath, forStatement, parent.getDepth() + 1, CodeElementType.FOR_STATEMENT);
+            CompositeStatementObject child = new CompositeStatementObject(ktFile, filePath,
+                forStatement, parent.getDepth() + 1, CodeElementType.ENHANCED_FOR_STATEMENT);
             parent.addStatement(child);
             KtExpression ktDeclaration = forStatement.getDestructuringDeclaration();
+            VariableDeclaration variableDeclaration = new VariableDeclaration(ktFile, filePath, forStatement.getLoopParameter());
+            child.addVariableDeclaration(variableDeclaration);
+            AbstractExpression abstractEx = new AbstractExpression(ktFile, filePath, forStatement.getLoopParameter(),
+                CodeElementType.ENHANCED_FOR_STATEMENT_PARAMETER_NAME);
+            child.addExpression(abstractEx);
             if (ktDeclaration != null) {
                 KtExpression initializer = forStatement.getDestructuringDeclaration().getInitializer();
```

Comment:
Please, use `ktDeclaration.getInitializer()` instead of` forStatement.getDestructuringDeclaration(`)

Reply:
Thank you, Zarina! It has been resolved.